In [6]:
# menampilkan dataset menggunakan library pandas
import pandas as pd
dataset = pd.read_csv('D:\MITA FILE\Semester 4\Kecerdasan Buatan\Sample_data\milknew.csv')
print(dataset)

       pH  Temprature  Taste  Odor  Fat   Turbidity  Colour  Grade
0     6.6          35      1     0     1          0     254      3
1     6.6          36      0     1     0          1     253      3
2     8.5          70      1     1     1          1     246      1
3     9.5          34      1     1     0          1     255      1
4     6.6          37      0     0     0          0     255      2
...   ...         ...    ...   ...   ...        ...     ...    ...
1054  6.7          45      1     1     0          0     247      2
1055  6.7          38      1     0     1          0     255      3
1056  3.0          40      1     1     1          1     255      1
1057  6.8          43      1     0     1          0     250      3
1058  8.6          55      0     1     1          1     255      1

[1059 rows x 8 columns]


In [7]:
# menampilkan kolom-kolom pada dataset
dataset.columns

Index(['pH', 'Temprature', 'Taste', 'Odor', 'Fat ', 'Turbidity', 'Colour',
       'Grade'],
      dtype='object')

Membagi data (x) dan target (y)

In [8]:
x = dataset[['pH', 'Temprature', 'Taste', 'Odor', 'Fat ', 'Turbidity', 'Colour']]
print(x)

       pH  Temprature  Taste  Odor  Fat   Turbidity  Colour
0     6.6          35      1     0     1          0     254
1     6.6          36      0     1     0          1     253
2     8.5          70      1     1     1          1     246
3     9.5          34      1     1     0          1     255
4     6.6          37      0     0     0          0     255
...   ...         ...    ...   ...   ...        ...     ...
1054  6.7          45      1     1     0          0     247
1055  6.7          38      1     0     1          0     255
1056  3.0          40      1     1     1          1     255
1057  6.8          43      1     0     1          0     250
1058  8.6          55      0     1     1          1     255

[1059 rows x 7 columns]


x pada dataset tersebut yaitu kolom pH, Temprature, Taste, Odor, Fat, Turbidity dan Colour

In [9]:
y = dataset['Grade']
print(y)

0       3
1       3
2       1
3       1
4       2
       ..
1054    2
1055    3
1056    1
1057    3
1058    1
Name: Grade, Length: 1059, dtype: int64


y pada dataset tersebut adalah kolom Grade

Membagi data train dan data set.

Pembagian data yaitu 80/20. Jumlah sample : 1059
Data train : 80%, data test : 20% 
Pembagian ini karena memberikan keseimbangan yang baik antara data train dan test

In [10]:
import sklearn
from sklearn.model_selection import train_test_split

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=1)

Dari syntax diatas, data test-nya yaitu 0.2 dari 1059 atau  20% dari 1059. Untuk melihat panjang dari x_test dan y_train yaitu sebagai berikut

In [11]:
len(x_test)

212

In [12]:
len(y_train)

847

Setelah membagi data menjadi data train dan data test, tahap selanjutnya yaitu membuat pemodelan Desicion tree dengan menggunakan library Scikit learn sebagai berikut


In [13]:
#Pemodelan 
from sklearn import tree

clf = tree.DecisionTreeClassifier()

Setelah membuat pemodelan, tahap selanjutnya yaitu Cross Validation untuk mengetahui performa dari model machine yang telah dibuat


In [17]:
# Tahap Cross validation
from sklearn.model_selection import cross_val_score

scores = cross_val_score(clf, x, y, cv=5)

Pada tahap ini, cross val score mengembalikan nilai berupa array yang terdiri dari tingkat akurasi pengujian 
setiap fold dataset.


In [18]:
scores

array([0.98113208, 1.        , 0.99056604, 1.        , 0.98578199])

Secara umum, jika hasil dari cross validation  memiliki nilai yang bervariasi dari 0.85 sampai 0.99 maka model tersebut 
dapat dikatakan baik. 
Hasil dari cross val score diatas, menunjukkan nilai array pertama yaitu 0.96.. dan array terakhir adalah 1, sehingga dapat disimpulkan bahwa model dari dataset ini adalah baik.


In [19]:
mean_score = scores.mean()
print(mean_score)

0.9914960207457749


Mean scores untuk menampilkan rata-rata skor dari semua fold, yang memberikan gambaran keseluruhan tentang kinerja model 